src/apps/myaccount/submodules/servicePlan/servicePlan.js

```diff
@@ -5,49 +5,109 @@ define(function(require) {
-				return template;
-			};
-			self.servicePlanRequestGetSummary({
-				success: function(result) {
-					var template = initTemplate(result);
-					args.hasOwnProperty('callback') && args.callback(template);
+
+			self._servicePlanGetTemplate({
+				callback: function($template) {
+					monster.pub('myaccount.renderSubmodule', $template);
+					args.hasOwnProperty('callback') && args.callback($template);
+				}
+			});
+		},
+
+		/**
+		 * @param  {Object} args
+		 * @param  {Object} [args.accountId]
+		 * @param  {Object} [args.servicePlan]
+		 * @param  {Boolean} [args.allowActions=true]
+		 * @param  {Function} args.callback
+		 */
+		_servicePlanGetTemplate: function(args) {
+			var self = this,
+				callback = args.callback,
+				accountId = _.get(args, 'accountId'),
+				servicePlan = _.get(args, 'servicePlan'),
+				allowActions = _.get(args, 'allowActions', true),
+				initTemplate = function initTemplate(dataToTemplate) {
+					var $template = $(self.getTemplate({
+						name: 'layout',
+						data: dataToTemplate,
+						submodule: 'servicePlan'
+					}));
+					return $template;
+				},
+				formatDataToTemplate = function formatDataToTemplate(result) {
+					var formattedServicePlan = self.servicePlanFormat(result),
+						invoices = formattedServicePlan.invoices;
+
+					return _.merge({
+						hasServicePlan: !_.isEmpty(invoices),
+						showActions: allowActions,
+						showBookkeeper: _.size(invoices) > 1,
+						showDueDate: !_.isUndefined(formattedServicePlan.dueDate)
+					}, formattedServicePlan);
+				};
+
+			monster.waterfall([
+				function(cb) {
+					if (!_.isUndefined(servicePlan)) {
+						return cb(null, servicePlan);
+					}
+					self.servicePlanRequestGetSummary({
+						data: {
+							accountId: accountId
+						},
+						success: function(result) {
+							cb(null, result);
+						}
+					});
+				}
+			], function(err, result) {
+				var funcs = [formatDataToTemplate, initTemplate],
+					getTemplate;
+				if (allowActions) {
+					funcs.push(self.servicePlanBindEvents.bind(self));
 				}
+				getTemplate = _.flow(funcs);
+				callback(getTemplate(result));
 			});
 		},
 
-		servicePlanFormat: function(data) {
+		/**
+		 * @param  {Object} servicePlan
+		 * @param  {Object[]} servicePlan.invoices
+		 * @return {Object}
+		 */
+		servicePlanFormat: function(servicePlan) {
 			var self = this;
+
 			return {
-				showBookkeeper: _.size(data.invoices) > 1,
-				hasSubscriptions: _.has(data, 'billing_cycle.next'),
-				hasServicePlan: !_.isEmpty(data.invoices),
-				totalAmount: _.reduce(data.invoices, function(acc, invoice) {
-					_.forEach(invoice.items, function(item) {
-						acc += item.total;
-					});
-					return acc;
-				}, 0),
-				dueDate: _.has(data, 'billing_cycle.next')
-					? monster.util.toFriendlyDate(data.billing_cycle.next, 'date')
-					: '?',
-				invoices: _.map(data.invoices, function(invoice, index) {
+				totalAmount: _
+					.chain(servicePlan.invoices)
```

Comment:
`chain` does not seem to be necessary in this case, because the only operation in the chain is `sumBy`.